crates/shared/src/order_validation.rs

```diff
@@ -495,6 +509,20 @@ impl OrderValidating for OrderValidator {
             _ => OrderClass::Ordinary,
         };
 
+        let num_limit_orders = self
+            .limit_order_counter
+            .count(owner)
+            .await
+            .map_err(ValidationError::Other)?;
```

Comment:
Counting the number of existing limit orders is only required if the user tries to create a limit order. I think this would also make the code more succinct.